org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/extensions/xsi/XSIFormatterTest.java

```diff
@@ -0,0 +1,121 @@
+				"<beans\r\n" + //
+				"    xmlns=\"http://www.springframework.org/schema/beans\"\r\n" + //
+				"    xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\"\r\n" + //
+				"    xmlns:util=\"http://www.springframework.org/schema/util\"\r\n" + //
+				"    xsi:schemaLocation=\"\r\n" + //
+				"        http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans.xsd\r\n"
+				+ //
+				"        http://www.springframework.org/schema/util http://www.springframework.org/schema/util/spring-util.xsd\">\r\n"
+				+ //
+				"\r\n" + //
+				"</beans>";
+		String expected = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\r\n" + //
+				"<beans\r\n" + //
+				"    xmlns=\"http://www.springframework.org/schema/beans\"\r\n" + //
+				"    xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\"\r\n" + //
+				"    xmlns:util=\"http://www.springframework.org/schema/util\"\r\n" + //
+				"    xsi:schemaLocation=\"http://www.springframework.org/schema/beans\r\n" + //
+				"                        http://www.springframework.org/schema/beans/spring-beans.xsd\r\n" + //
+				"                        http://www.springframework.org/schema/util\r\n" + //
+				"                        http://www.springframework.org/schema/util/spring-util.xsd\">\r\n" + //
+				"\r\n" + //
+				"</beans>";
+		assertFormat(content, expected, settings);
+	}
+
+	@Test
+	public void xsiSchemaLocationSplitOnLocation() throws BadLocationException {
+		SharedSettings settings = createSettings();
+		XSISchemaLocationSplit.setSplit(XSISchemaLocationSplit.onLocation, settings.getFormattingSettings());
+		String content = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\r\n" + //
+				"<beans\r\n" + //
+				"    xmlns=\"http://www.springframework.org/schema/beans\"\r\n" + //
+				"    xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\"\r\n" + //
+				"    xmlns:util=\"http://www.springframework.org/schema/util\"\r\n" + //
+				"    xsi:schemaLocation=\"\r\n" + //
+				"        http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans.xsd\r\n"
+				+ //
+				"        http://www.springframework.org/schema/util http://www.springframework.org/schema/util/spring-util.xsd\">\r\n"
+				+ //
+				"\r\n" + //
+				"</beans>";
+		String expected = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\r\n" + //
+				"<beans\r\n" + //
+				"    xmlns=\"http://www.springframework.org/schema/beans\"\r\n" + //
+				"    xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\"\r\n" + //
+				"    xmlns:util=\"http://www.springframework.org/schema/util\"\r\n" + //
+				"    xsi:schemaLocation=\"http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans.xsd\r\n"
+				+ //
+				"                        http://www.springframework.org/schema/util http://www.springframework.org/schema/util/spring-util.xsd\">\r\n"
+				+ //
+				"\r\n" + //
+				"</beans>";
+		assertFormat(content, expected, settings);
+	}
+

```

Comment:
Maybe try a case with an empty `xsi:schemaLocation`. It worked as expected in my manual testing, but it might be good for regressions.

Reply:
fixed

Reply:
I have added too some test with tabs.